[BOJ]2805.py

```diff
@@ -0,0 +1,20 @@
+# 백준 - https://www.acmicpc.net/problem/2805
+
+import sys
+
+N, M = map(int, sys.stdin.readline().split())
+trees = list(map(int, sys.stdin.readline().split()))
+
+high = max(trees)
+low = 0
+min_h = 0
+
+while low <= high:
+    mid = (low + high) // 2
+    if sum([tree - mid for tree in trees if (tree - mid) > 0]) >= M:
+        min_h = mid
+        low += 1
```

Comment:
```suggestion
        low = mid +1
```
슬라이딩 윈도우 잘 보았습니다 ㅋㅋㄹㅃㅃ